apps/multisig/src/domains/chains/extrinsics.ts

```diff
@@ -419,6 +420,180 @@ export const useAsMulti = (extensionAddress: Address | undefined, t?: Transactio
   return { asMulti, ready: ready && !!estimatedFee, estimatedFee }
 }
 
+export const useAsMultiThreshold1 = (
+  extensionAddress: Address | undefined,
+  hash: `0x${string}` | undefined,
+  t?: Transaction
+) => {
+  const [estimatedFee, setEstimatedFee] = useState<Balance | undefined>()
+  const [selectedMultisig] = useSelectedMultisig()
+  const multisig = useMemo(() => t?.multisig ?? selectedMultisig, [selectedMultisig, t?.multisig])
+  const { api } = useApi(multisig.chain.genesisHash)
+  const insertTxMetadata = useInsertTxMetadata()
+  const { addContract } = useAddSmartContract()
+  const injectMetadata = useInjectMetadata(multisig?.chain.genesisHash ?? supportedChains[0]!.genesisHash)
+  const nativeToken = useRecoilValueLoadable(tokenByIdQuery(multisig.chain.nativeToken.id))
+  const setRawPendingTransactionDependency = useSetRecoilState(rawPendingTransactionsDependency)
+
+  const extrinsic = useMemo(() => {
+    if (!api || !t?.callData) return undefined
+    return decodeCallData(api, t.callData)
+  }, [api, t?.callData])
+
+  const ready = useMemo(
+    () =>
+      !!api && extensionAddress && !!extrinsic && nativeToken.state === 'hasValue' && !!hash && multisig !== undefined,
+    [api, extensionAddress, extrinsic, hash, multisig, nativeToken.state]
+  )
+
+  // Creates some tx from calldata
+  const createExtrinsic = useCallback(async () => {
+    if (!ready || !api || !extrinsic || !extensionAddress) return
+
+    if (!api.tx.multisig?.asMultiThreshold1) throw new Error('chain missing multisig pallet')
+
+    return api?.tx.multisig.asMultiThreshold1(
+      Address.sortAddresses(multisig.signers)
+        .filter(s => s && !s.isEqual(extensionAddress))
+        .map(s => s.bytes),
+      extrinsic.method.toHex()
+    )
+  }, [api, extensionAddress, extrinsic, multisig.signers, ready])
```

Comment:
seems like nothing in here needs to be awaited, we can just create this as a useMemo so then we dont have to invoke this everytime we use it, just asMultiThreshold1Extrinsic.something